Tools/CISupport/ews-build/steps.py

```diff
@@ -2150,6 +2174,188 @@ def hideStepIf(self, results, step):
+
+        defer.returnValue([list_of_prs, passed_status_check, failed_status_check])
+
+
+class SafeMergeToMergeQueue(buildstep.BuildStep, GitHubMixin, AddToLogMixin):
+    name = 'safe-merge-to-merge-queue'
+    flunkOnFailure = False
+    haltOnFailure = False
+    LABELS_TO_REMOVE = [GitHubMixin.SAFE_MERGE_QUEUE_LABEL]
+
+    @defer.inlineCallbacks
+    def run(self):
+        safe_prs = self.getProperty('safe_pr_list')
+        pr_number = safe_prs.pop()
+        yield self._addToLog('stdio', 'Updating labels for PR {}...\n'.format(pr_number))
+        self.setProperty('safe_pr_list', safe_prs)
+        if len(safe_prs):
+            self.build.addStepsAfterCurrentStep([SafeMergeToMergeQueue()])
+        rc = yield self.update_labels(pr_number)
+        defer.returnValue(rc)
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Labelled safe PRs with merge-queue."}
+        elif self.results == FAILURE:
+            return {'step': f"Failed to label safe PRs with merge-queue."}
+        return buildstep.BuildStep.getResultSummary(self)
+
+    @defer.inlineCallbacks
+    def update_labels(self, pr_number):
+        repository_url = self.getProperty('repository', '')
+
+        did_remove_labels = yield self.remove_labels(pr_number, self.LABELS_TO_REMOVE, repository_url=repository_url)
+        if did_remove_labels:
+            yield self._addToLog('stdio', 'Successfully removed existing labels from PR {}.\n'.format(pr_number))
+        else:
+            yield self._addToLog('stdio', 'Failed to remove existing labels from PR {}.\n'.format(pr_number))
+
+        did_add_label = yield self.add_label(pr_number, self.SAFE_MERGE_QUEUE_LABEL, repository_url=repository_url)
+        if did_add_label:
+            yield self._addToLog('stdio', 'Successfully added merge-queue label to PR {}.\n'.format(pr_number))
+        else:
+            yield self._addToLog('stdio', 'Failed to add merge-queue label to PR {}.\n'.format(pr_number))
+        defer.returnValue(SUCCESS if did_remove_labels and did_add_label else FAILURE)
+
+
+class SafeMergeToMergeBlocked(buildstep.BuildStep, GitHubMixin, AddToLogMixin):
+    name = 'safe-merge-to-merge-blocked'
+    flunkOnFailure = False
+    haltOnFailure = False
+    LABELS_TO_REMOVE = [GitHubMixin.SAFE_MERGE_QUEUE_LABEL]
+
+    @defer.inlineCallbacks
+    def run(self):
+        failed_prs = self.getProperty('blocked_pr_list', [])
+        pr_number = failed_prs.pop()
+        yield self._addToLog('stdio', 'Updating labels for PR {}...\n'.format(pr_number))
+        self.setProperty('blocked_pr_list', failed_prs)
+        if len(failed_prs):
+            self.build.addStepsAfterCurrentStep([SafeMergeToMergeBlocked()])
+        rc = yield self.update_labels(pr_number)
+        defer.returnValue(rc)
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Labelled blocked PRs with merge-blocked."}
```

Comment:
Nit: this step seems to be labelling a single PR, not `PRs`. Can add the PR number in this message.